test/test_circuitzoo_purification.jl

```diff
@@ -34,4 +43,86 @@ const bell = StabilizerState("XX ZZ")
             end
         end
     end
+
+    for rep in [QuantumOpticsRepr]
+        for leaveout in [:X, :Y, :Z]
+            r = Register(4, rep())
+            rnd = rand() / 4 + 0.5
+            noisy_pair = noisy_pair_func(rnd)
+            initialize!(r[1:2], noisy_pair)
+            initialize!(r[3:4], noisy_pair)
+            if Purify2to1(leaveout)(r[1:4]...)==true
+                @test abs(observable(r[1:2], projector(bell))) >= rnd
+            end
+        end
+    end
+end
+
+@testset "3to1" begin
+    for rep in [QuantumOpticsRepr, CliffordRepr]
+        for fixtwice in [:X, :Y, :Z]
+            # test that pure state gets mapped to pure state
+            r = Register(6, rep())
+            initialize!(r[1:6], bell⊗bell⊗bell)
+            @test Purify3to1(fixtwice)(r[1], r[2], [r[3], r[5]], [r[4], r[6]])==true
+            @test observable(r[1:2], projector(bell))≈1.0
+            # TODO: Should also taget qubits 1 and 2
+            for error in [:X, :Y, :Z], target in 3:6
+                r = Register(6, rep())
+                for i in 1:3
+                    initialize!(r[(2*i-1):(2*i)], bell)
+                end
+                apply!(r[target], Dict(:X=>X, :Y=>Y, :Z=>Z)[error])
+                @test Purify3to1(fixtwice)(r[1], r[2], [r[3], r[5]], [r[4], r[6]])==false
+            end
+            ## When [error, fixtwice] in {[X,Z], [Z,Y], [Y,X]} it yields true is that supposed to happen?
```

Comment:
Could you elaborate, what does that mean?

Reply:
Well apparently it does not detect the error when [error, fixtwice] are in {[X,Z], [Z,Y], [Y,X]}. It seems a quite consistent behaviour and i think it's supposed to happen but i dont seem to figure out why

Reply:
I ran it and the double selection circuits do have better fidelity then the ordered ones. 
The notebook is in a repo in my profile

Reply:
https://github.com/Barosandu/JLNotebookQuantum